process/block/argProcessor.go

```diff
@@ -99,6 +99,7 @@ type ArgBaseProcessor struct {
 	OutGoingOperationsPool         OutGoingOperationsPool
 	DataCodec                      sovereign.DataCodecProcessor
 	TopicsChecker                  sovereign.TopicsChecker
+	OperationsHasher               hashing.Hasher
```

Comment:
This approach is not correct.
You should not pass this interface everywhere from factories up to baseProcess and metaBlock process which will never use this.

This is the reason why you have `sovereignBlockProcessorFactory`. That factory should create the hasher based on a config string. Max you can do is to add `OperationHasherType` as `string` in `BaseArgs` and only make use of it inside sovereign factory.

Also, you will have to delete the disabled operation hasher

Reply:
:+1: also the other members of ArgBaseProcessor should be used just where they are needed